From 671b4ff1b1140027b81f892725127075fd121dc5 Mon Sep 17 00:00:00 2001
From: Potin Lai <potin.lai.pt@gmail.com>
Date: Mon, 3 Feb 2025 20:26:28 +0800
Subject: [PATCH] bindings: ipmi: ssif-bmc: Add property to adjust response
 timeout

The SSIF response is handled by a userspace application, which may exceed
the default 500 ms timeout when the BMC or application is under heavy load.

Add a new device tree property, `timeout-ms`, allowing platforms to adjust
the response timeout based on their specific requirements. This helps
prevent timeouts in scenarios where longer processing times are expected.

Signed-off-by: Potin Lai <potin.lai.pt@gmail.com>

LORE: https://lore.kernel.org/all/20250203-ssif-response-timeout-ms-v1-1-6fe279740181@gmail.com/
---
 Documentation/devicetree/bindings/ipmi/ssif-bmc.yaml | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/Documentation/devicetree/bindings/ipmi/ssif-bmc.yaml b/Documentation/devicetree/bindings/ipmi/ssif-bmc.yaml
index 02b662d78..c8a863d44 100644
--- a/Documentation/devicetree/bindings/ipmi/ssif-bmc.yaml
+++ b/Documentation/devicetree/bindings/ipmi/ssif-bmc.yaml
@@ -19,6 +19,12 @@ properties:
   reg:
     maxItems: 1
 
+  timeout-ms:
+    minimum: 0
+    maximum: 4294967295
+    default: 500
+    description: Response timeout in millisecond.
+
 required:
   - compatible
   - reg
-- 
2.34.1

